recipes/cusparselt/conda_build_config.yaml

```diff
@@ -0,0 +1,18 @@
+docker_image:                           # [os.environ.get("BUILD_PLATFORM", "").startswith("linux") or (os.environ.get("CONFIG_VERSION", "1") == "1" and linux)]
+   - quay.io/condaforge/linux-anvil-comp7        # [os.environ.get("BUILD_PLATFORM") == "linux-64" or (os.environ.get("CONFIG_VERSION", "1") == "1" and linux64)]
+ 
+   - quay.io/condaforge/linux-anvil-cos7-cuda:9.2     # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+   - quay.io/condaforge/linux-anvil-cos7-cuda:10.0    # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+   - quay.io/condaforge/linux-anvil-cos7-cuda:10.1    # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+   - quay.io/condaforge/linux-anvil-cos7-cuda:10.2    # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+   - quay.io/condaforge/linux-anvil-cuda:11.0         # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+
+cuda_compiler:   # [linux64 or win]
+  - nvcc         # [linux64 or win]
+cuda_compiler_version:
+  - None
+  - 9.2                        # [linux64]
+  - 10.0                       # [linux64 or win]
+  - 10.1                       # [linux64 or win]
+  - 10.2                       # [linux64 or win]
+  - 11.0                       # [linux64 or win]
```

Comment:
Please check out https://github.com/conda-forge/cutensor-feedstock/blob/master/recipe/conda_build_config.yaml. Conda-Forge zips 4 keys together: `cudnn`, `cuda_compiler_version`, `docker_image` and `cdt_name`. So if we are implementing a custom `conda_build_config.yaml` to overwrite the default setting in `conda-forge-pinning-feedstock`, we nee to be explicit on allowed keys.

Reply:
In the case of cuSPARSELt, it only works with 11.x, and the next release will require 11.2. 
Also, I'm guessing we don't need `docker_image`, `cudnn`, and `cdt_name`???

Reply:
It is fine leaving all available CUDA versions here, as the actual skip is done in `meta.yaml`. But whatever `docker_image` we list here, they must be accompanied by the rest of the 3 keys. The key-zipping is done globally, not per feedstock, so it is not something controllable here. This was why you got this error in Azure:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=291111&view=logs&j=232b7f5e-8d64-5438-fada-8478f379d226&t=e8516d7e-1153-5090-5807-53ae833fddd8&l=557

Reply:
What "keys" are you referring to??

Reply:
By keys I am referring to
> `cudnn`, `cuda_compiler_version`, `docker_image` and `cdt_name`

and their values follow with a `-` prefix, say,
```
cuda_compiler_version:
  - None
  - 9.2
  - 10.0
  - ...
``` 
This is what's done in conda-forge-pinning-feedstock:
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/a365475270cfefdd44975873091d6ad7503b7e9a/recipe/conda_build_config.yaml#L141-L145
which affects globally (i.e. all Conda-Forge packages). 

If any of these keys appear in `conda_build_config.yaml` (which we normally don't need, as the default is to inherit the setting from CF pinning), they must appear together so that they can be zipped correctly by conda-smithy. This is why this file in cuTENSOR/cuDNN/NCCL looks so complicated. I think it'd work if you just copy this file from the cuTENSOR feedstock, to which I linked earlier.

Reply:
Yeah I think we can drop this file. Though please correct me if I'm missing something

Reply:
I think that'd also work!

Reply:
Okay, so I copied the file from cuTENSOR feedstock but removed all references to ppc64le. We don't plan to support Power9

Reply:
Matt, the 3 keys are still missing. Let's try just removing this file entirely as John suggested.

Reply:
Thanks for the context Matt. Yeah in that case we don't need the file. Could you please drop it?

Reply:
Adding a bit more context in the hope that Matt wouldn't get confused by me 😄 

We have this file `conda_build_config.yaml` in other feedstocks because 1. we wanted to span the build matrix beyond the default setting in conda-forge-pinning, and 2. for certain cases we want to use CentOS 7 docker images, mostly due to the requirement of using a newer glibc version in some packages (such as cuTENSOR).

In the case of cuSPARSELt, if we're only building on CUDA 11.0 and supporting 11.x, none of the above applies. On Conda-Forge (and likely elsewhere), 11.0 is enabled by default, and all CUDA 11.x images are already on cos7.

Reply:
Thanks @leofang @jakirkham! I'll go ahead and remove the file